src/main/java/org/broadinstitute/hellbender/engine/SlidingWindowWalker.java

```diff
@@ -0,0 +1,185 @@
+package org.broadinstitute.hellbender.engine;
+
+import htsjdk.samtools.SAMSequenceDictionary;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.engine.filters.CountingReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.engine.filters.WellformedReadFilter;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * An SlidingWindowWalker is a tool that processes a single window over the genome/intervals at a time, with the ability
+ * to query optional overlapping sources of reads, reference data, and/or variants/features.
+ *
+ * The windows are constructed over for each interval with a concrete window and step size, and padding overlapping
+ * sources if requested
+ *
+ * SlidingWindow authors must implement the {@link #apply}, {@link #getWindowSize}, {@link #getWindowStep} and {@link
+ * #getWindowPadding} methods, and may optionally implement {@link #onTraversalStart} and/or {@link #onTraversalSuccess}.
+ *
+ * @author Daniel Gómez-Sánchez (magicDGS)
+ */
+public abstract class SlidingWindowWalker extends GATKTool {
+
+    @Argument(fullName = "disable_all_read_filters", shortName = "f", doc = "Disable all read filters", common = false, optional = true)
+    public boolean disableAllReadFilters = false;
+
+    protected final static String WINDOW_SIZE_ARGUMENT = "windowSize";
+
+    protected final static String WINDOW_STEP_ARGUMENT = "windowStep";
+
+    protected final static String WINDOW_PADDING_ARGUMENT = "windowPadding";
```

Comment:
There's currently no way to hide `@Arguments` directly, but you could effectively do this via `ArgumentCollections`. See, eg., the way `GATKTool` uses `ArgumentCollections` to select between required and optional versions of the same arguments (such as `RequiredReadInputArgumentCollection` vs. `OptionalReadInputArgumentCollection`). You could do the same thing in your `SlidingWindowWalker` to allow tool authors to turn different arguments on and off, by having an "empty" argument collection for each sliding window argument.

This is a bit ugly, but I still think it's better to have the arguments controlling traversal defined in the traversal itself rather than in the tools.